models/classes/webhooks/task/WebhookTask.php

```diff
@@ -103,44 +101,20 @@ private function performRequest(RequestInterface $request, WebhookAuthInterface
         try {
             $response = $this->getWebhookSender()->performRequest($request, $authConfig);
         } catch (BadResponseException $badResponseException) {
-            return $this->handleBadResponseException($badResponseException);
+            if ($badResponseException instanceof ServerException) {
```

Comment:
Why we can't catch this error and perform this action?

Reply:
1. that way you have to create temp variable, record report, retryTask(), return variable (less readable, can cause CodeClimate complaining about lines count in method)
2. in case of async queue (which should be used with webhooks) the order doesn't matter
3. in case of sync queue we perform retryTask() before returning report anyway

But I can change it

Reply:
I don't understand this. 
1. Why do you have to create temporary variables? What kind of variables? Don't get it. 
```suggestion
          catch(ServerException $exception) {
              $this->retryTask();
          }
```

Reply:
no, it should be (as it was before) a branch in handling BadResponseException and `return $this->getWebhookTaskReports()->reportBadResponseException(...)` should still be called

Reply:
Why can't it be caught in another catch?

Reply:
This if looks silly here.

Reply:
It can, I see no difference here

Reply:
As I said. this `if` statement looks not right here?

Reply:
fixed